### DISCUSSION TITLE
[RFC] Expose worker's metrics to the `metrics` plugin if available.

### DISCUSSION THREAD

**Discussion by rustatian** [roadrunner-server/roadrunner#642] [Ideas]
Discussed in our Discord server.
An interesting proposal was from the @alex88 . Expose our worker's metrics such as CPU, Memory, Execs, Created via the `metrics` plugin. These metrics can be exposed to the grafana with the nice charts or used in the Kubernetes engine.
Can be done via the `server` -> `metrics` Collects call. So, `server` plugin can provide inside the pool/worker some method (??) to add metrics and have control over them.

**Comment by rustatian** [comment]
User will see smt like:
```
# TYPE pool metrics
worker_pid execs 1000
worker_pid memory_in_use 101.5
```
And so on

**Comment by alex88** [comment]
I just wanted to add another thing. My use case is mainly focused on autoscaling the kubernetes pods running the php application to avoid queuing requests and scaling the pods when most of the workers are busy.
We're currently using php-fpm and exporting prometheus metrics using a tool that exposes these metrics https://github.com/hipages/php-fpm_exporter#metrics-collected and we use the active/idle/total processes count to autoscale.
It would be great to have these metrics available on the metrics plugin too so we can use those metrics to do the same with RR.
